flagscale/train/hetero/parallel_context.py

```diff
@@ -586,19 +595,25 @@ def _backtrack(mesh_index, prev_rank, path):
             ranks_list = current_mesh.get_all_process_group_ranks("pp")
             valid_ranks_list = []
             for ranks in ranks_list:
-                if prev_rank == -1 or (prev_rank, ranks[0]) in self._inter_mesh_process_groups:
+                mesh_is_connect = False
+                for prev_path_ranks in path:
+                    for prev_path_rank in prev_path_ranks:
+                        if (prev_path_rank, ranks[0]) in self._inter_mesh_process_groups:
+                            mesh_is_connect = True
```

Comment:
Change mesh_is_connect to mesh_is_connected.

Reply:
OK.